content/what-is-cross-origin-resource-sharing/index.md

```diff
@@ -0,0 +1,166 @@
+-   `Origin` - Where the request is coming from. For us, that's our previous domain - `emailpassword.demo.supertokens.com`
+
+After getting this pre-flight / `OPTIONS` request, the API server sends over a pre-flight response. Here's our response from `api-emailpassword.demo.supertokens.com`.
+
+```
+HTTP/1.1 204 No Content
+Access-Control-Allow-Origin: https://emailpassword.demo.supertokens.com/
+Access-Control-Allow-Credentials: true
+Access-Control-Allow-Methods: GET,PUT,POST,DELETE
+Access-Control-Allow-Headers: content-type,rid,fdi-version,anti-csrf
+```
+
+Let's break this response down.
+
+-   `Access-Control-Allow-Origin` - The origins that the API server has whitelisted
+-   `Access-Control-Allow-Credentials` - The server telling us whether the actual request can include cookies in it, or that the response of the actual request can set-cookies. In our case, cookies refer to the session tokens of the user, which act as the credentials of the user once they're signed in.
+-   `Access-Control-Allow-Methods` - A comma-separated list of HTTP methods that the API domain allows for cross-origin requests
+-   `Access-Control-Allow-Headers` - A comma-separated list of HTTP headers that the API domain allows for cross-origin requests
+
+The browser then takes this response from the API server to determine if the actual request should be sent. If the response from the API doesn't include the requested origin, methods, or headers from the preflight request, then the browser will not send the actual request.
+
+## CORS actual request
+
+If the response from the API includes the requested origin, it's time to send the actual `POST` request to sign in.
+
+```
+POST /auth/signin HTTP/1.1
+Host: http://api-emailpassword.demo.supertokens.com/content-type: application/json
+fdi-version: 1.15
+rid: emailpassword
+Content-Length: 92
+Origin: https://emailpassword.demo.supertokens.com/
+```
+
+Note that this request includes both the `Host` and `Origin` headers. In addition, we see the content-type header show up in the request (near the end of the `Host` request line). Looking back to the pre-flight response, we can see that content-type is one of the allowed headers.
+
+Now let's take a look at the response from the server.
+
+```
+HTTP/1.1 200 OK
+Access-Control-Allow-Origin: https://emailpassword.demo.supertokens.com/
+Access-Control-Allow-Credentials: truefront-token: ...
+Access-Control-Expose-Headers: front-token, id-refresh-token
+Set-Cookie: ...
+```
+
+Here, we get the response from the server with cookies and tokens that allows us to proceed with the sign-in operation. One thing to note is that compared to our pre-flight, we now also have an additional `Access-Control-Expose-Headers` header.
+
+- `Access-Control-Expose-Headers` - The server indicating which response headers are available to scripts running in the browser.
```

Comment:
```suggestion
- `Access-Control-Expose-Headers` - The server indicating which response headers are available for the scripts running in the browser. Interestingly, the reason this exists is not because of security, but because of backward compatibility to when CORS did not exist.
```